src/components/chip/chip.tsx

```diff
@@ -0,0 +1,279 @@
+     * ID of the chip. Must be unique.
+     */
+    @Prop()
+    id: number | string;
+
+    /**
+     * Label displayed on the chip
+     */
+    @Prop()
+    public text: string;
+
+    /**
+     * Icon of the chip.
+     */
+    @Prop()
+    public icon: string | Icon;
+
+    /**
+     * If supplied, the chip will become a clickable link.
+     */
+    @Prop()
+    public link?: Omit<Link, 'text'>;
+
+    /**
+     * The value of the badge, displayed on the chip.
+     */
+    @Prop()
+    public badge?: string | number;
+
+    /**
+     * Set to `true` to disable the chip.
+     */
+    @Prop({ reflect: true })
+    public disabled = false;
+
+    /**
+     * Set to `true` to render the chip as a static UI element.
+     * Useful when the parent component has a `readonly` state.
+     */
+    @Prop({ reflect: true })
+    public readonly = false;
+
+    /**
+     * Set to `true` to visualize the chip in a "selected" state.
+     * This is typically used when the chip is used in a chip-set
+     * along with other chips.
+     */
+    @Prop({ reflect: true })
+    public selected = false;
+
+    /**
+     * Set to `true` to render a remove button on the chip.
+     */
+    @Prop({ reflect: true })
+    public removable = false;
+
+    /**
+     * Set to `filter` to render the chip with a distinct style
+     * suitable for visualizing filters.
+     */
+    @Prop({ reflect: true })
+    public type: 'filter';
+
+    /**
+     * Fired when clicking on buttons (not links!)
+     * inside the breadcrumbs.
+     */
+    @Event()
+    public remove: EventEmitter<Chip>;
+
+    @Element()
+    private host: HTMLLimelChipElement;
+
+    private button: HTMLButtonElement;
+    private anchor: HTMLAnchorElement;
+
+    public componentWillLoad() {
+        makeEnterClickable(this.host as HTMLElement);
+    }
+
+    public disconnectedCallback() {
+        this.removeEnterClickable();
+    }
+
+    public render() {
+        if (this.link) {
+            return this.renderAsLink();
+        }
+
+        return this.renderAsButton();
+    }
+
+    private renderAsButton = () => {
+        return [
+            <button
+                class="chip"
+                role="button"
+                disabled={this.disabled || this.readonly}
+                onKeyDown={this.handleDeleteKeyDown}
+            >
+                {this.renderIcon()}
+                {this.renderLabel()}
+                {this.renderBadge()}
+            </button>,
+            this.renderRemoveButton(),
+        ];
+    };
+
+    private renderAsLink = () => {
+        return [
+            <a
+                class="chip"
+                href={this.link.href}
+                title={this.link.title}
+                aria-disabled={this.disabled || this.readonly}
+                tabindex={this.disabled ? -1 : 0}
+                onClick={(e) => {
+                    if (this.disabled) {
+                        e.preventDefault();
+                    }
+                }}
```

Comment:
```suggestion
```

Good point, this was also an over-engineering. The combination of the tab-index (the line before) and the CSS styles for `[disabled]` already do the job. What do you think if we remove it totally?

Reply:
Screen readers might still be able to focus the element… 🤔 (They _should_ be, since conveying that there's something there but that it's disabled is important, as far as I know.)

Reply:
Yes. I see. That's correct. But I just got a hallelujah idea. Can we define this on the `<Host>` instead?
```tsx
render() {
    return (
        <Host onClick={this.filterClick}>
            {/* rest of the component */}
        </Host>
    );
}
```

This way, we can remove the CSS rule 
```scss
:host(limel-chip[disabled]) { 
    pointer-events: none; 
}
```
, as the filterClick method will now handle preventing clicks when the component is disabled.

The reason I think this is better is that we have always used that CSS above, on components like `limel-button` or `limel-icon-button` or any similar component that the consumers will define an `onClick` on. This have always created a lot of problems, since when you set `pointer-events: none; ` you will lose every-wonderful-thing that you would get from pointers interacting with the component; e.g. `:focus` and or `:hover`, which could be so useful in communicating with users using micro interactions. Another thing you would lose is the possibility of attaching a `limle-tooltip` to the component. Because when it is `disabled`, the pointer-events would be none, which means you cannot hover the component to get a tooltip. The tooltip could be used smartly, to explain why the button is disabled for example, (or show the `title` or `label` of the icon-button, which is hard to know without hovering it).

If `<Host onClick={this.filterClick}>` will do the exact same thing (also handle the case both for link and button, then I can make follow up PRs in other components, and remove that super annoying `pointer-events: none; `.

Reply:
It seems like that ought to work, yes…

Sounds like a great idea! Please feel free to try it!

Reply:
Yeay 152cc5630a546154b121e0f3f82b0c0a662a135f it works like a charm! 
Look @LucyChyzhova ! Finally, we found a solution [this super annoying issue](https://github.com/Lundalogik/crm-feature/issues/3317).

😊 You remember this: 

https://github.com/Lundalogik/lime-elements/blob/e1c555dabc0084927d39ec6755ccde86b3360512/src/design-guidelines/disabled-hidden/examples/disabled-vs-hidden.tsx#L83C47-L83C47